src/synapse/synapse_handler.rs

```diff
@@ -32,16 +32,24 @@ pub async fn accept_room_invitation<'a>(
     room_message_body: Option<&str>,
     synapse: &SynapseComponent,
 ) -> Result<(), CommonError> {
-    // Check if it's a `accept` event.
-    if room_event != FriendshipEvent::ACCEPT {
-        return Ok(());
+    let joined_rooms = synapse.get_joined_rooms(token).await;
```

Comment:
Should we avoid this call altogether and just join the room?

Reply:
we could, but the request of joining the room will return an error that we should filter out